esp-hal/src/spi/slave.rs

```diff
@@ -582,15 +589,22 @@ where
         Ok(tx.start_transfer()?)
     }
 
-    fn start_read_bytes_dma(&mut self, ptr: *mut u8, len: usize, rx: &mut RX) -> Result<(), Error> {
+    unsafe fn start_read_bytes_dma(
+        &mut self,
+        ptr: *mut u8,
+        len: usize,
+        rx: &mut RX,
+    ) -> Result<(), Error> {
         let reg_block = self.register_block();
 
         rx.is_done();
 
         self.enable_dma();
 
         reset_dma_before_load_dma_dscr(reg_block);
-        rx.prepare_transfer_without_start(false, self.dma_peripheral(), ptr, len)?;
+        unsafe {
```

Comment:
We shouldn't need an `unsafe` block inside a function that's already marked `unsafe`.

Reply:
currently yes .... https://rust-lang.github.io/rfcs/2585-unsafe-block-in-unsafe-fn.html

Ok will remove the unsafe blocks inside unsafe function

Reply:
Well that's certainly... something...

Reply:
Maybe we should decide if we want that lint in future to be prepared for the next edition but ... we currently get 235 warnings then - probably nothing to think about today 😆